bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Cursor.java

```diff
@@ -684,6 +680,16 @@ private static class ImageDataWithMaskCursorHandleProvider extends ImageDataCurs
 
 	public ImageDataWithMaskCursorHandleProvider(ImageData source, ImageData mask, int hotspotX, int hotspotY) {
 		super(source, hotspotX, hotspotY);
+		if (mask == null) {
+			if (source.getTransparencyType() != SWT.TRANSPARENCY_MASK) {
+				SWT.error(SWT.ERROR_NULL_ARGUMENT);
+			}
+			mask = source.getTransparencyMask();
```

Comment:
This changes the current behavior: now the provider is initialized with a mask retrieved from the source instead of just storing the originally passed mask (which may be null). Maybe this is a cleaner way of doing it but (1) it should then be done independent from this change of making handle creation lazy and (2) we then need to ensure that a mask that is first created from source and then scaled is the same as first scaling the source and then extracting a mask out of it.